_docs/cardano/2017-01-06-updating.md

```diff
@@ -157,93 +154,96 @@ modifications to the protocol with some degree of concurrency.
-transaction in a block with version 1. This is the point. We cannot make
-use of a restriction without waiting for the network to start assuming
-the old version to be deprecated (since we can only start rejecting blocks
-when their version is deprecated).
 
-Also, having `attributes` field, we may become attacked by someone
-stating he uses higher version of the protocol, polluting attributes
-with meaningless keys, and we would have to accept his block. For this
-reason, the blocks with the version x.y.z are rejected prior to
-application of the update with the protocol version x.y.z approved by
-stake.
-
-That said, we propose the following default rule for version stabilization:
-
- > if 95%  of the latest 2000 blocks contain the version `>= (maj, min,
- > alt)`, then reject all blocks with lower versions. Let’s call it the
- > soft fork stabilization point (or stabpoint).
+1. An old version of the client should always find recent blocks valid. (This is
+   what
+   [BIP-99](https://github.com/bitcoin/bips/blob/ed283b05b332b85b6fd683be3a5d73fab6c15554/bip-0099.mediawiki)
+   calls "everything invalid remains invalid").
+2. Some blocks issued by an old version of the client may be considered invalid
+   by a newer version.
+
+Obviously, imposing rule 2 as it is may cause the network to be split into two
+parts: a stakeholder with a stake large enough could update and maintain their
+own chain, rejecting blocks from others, but others would be still able to
+maintain their chain, rejecting blocks from this stakeholder (since he does not
+have a majority of stake, and therefore cannot catch up with the rest of the
+crowd, so his chain is shorter). A simple resolution rule could go like this: if
+95% of the latest 2016 blocks have a newer block version, the blocks with the
+older version are rejected.
+
+It may seem unclear why we would like to make some block version invalid at some
+moment. The key insight here is that a new feature is actually a restriction on
+what we had had previously. For example, currently we have plain old
+transactions which may contain either PublicKey-based addresses or Script-based
+ones. Then at some point we decide to include a third address type (no matter
+what its purpose is). Which strategy do we need for verifying a block with a
+transaction with an address of unknown type? Obviously the only option is not
+verifying this address.
+
+And then imagine somebody proposing a transaction to this address, possibly
+doing that with an intention to secure funds from being spent until some
+conditions are met — and then watching them being spent in some other
+transaction in a block with version 1. This is the point. We cannot make use of
+a restriction without waiting for the network to start assuming the old version
+to be deprecated (since we can only start rejecting blocks when their version is
+deprecated).
+
+Another problem is that in our implementations all blocks have a field called
```

Comment:
Can it be reformulated somehow? Reader can think this _another problem_ is about the same concern, i. e. that we need to deprecate old version. But this _another problem_ is about different thing. It's about that we need to prohibit usage of new features from new version until it's adopted.